cmd/lightclient/sentinel/proto/ssz_snappy/gossip.go

```diff
@@ -0,0 +1,79 @@
+package ssz_snappy
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"gfx.cafe/util/go/bytepool"
+
+	ssz "github.com/ferranbt/fastssz"
+	"github.com/golang/snappy"
+	"github.com/ledgerwatch/erigon/cmd/lightclient/sentinel/proto"
+	pubsub "github.com/libp2p/go-libp2p-pubsub"
+)
+
+type GossipCodec struct {
+	sub *pubsub.Subscription
+	top *pubsub.Topic
+	mu  sync.Mutex
+}
+
+func NewGossipCodec(
+	sub *pubsub.Subscription,
+	top *pubsub.Topic,
+) proto.GossipCodec {
+	return &GossipCodec{
+		sub: sub,
+		top: top,
+	}
+}
+
+// decode into packet p, then return the packet context
+func (d *GossipCodec) Decode(ctx context.Context, p proto.Packet) (sctx *proto.GossipContext, err error) {
+	sctx, err = d.readPacket(ctx, p)
+	return
+}
+
+func (d *GossipCodec) WritePacket(ctx context.Context, p proto.Packet) error {
+	if val, ok := p.(ssz.Marshaler); ok {
+		bts := bytepool.Get(val.SizeSSZ())
```

Comment:
please do it with the normal `make()` method do not use other external libraries for stuff like that